### ISSUE TITLE
Modifing a profile yields "unknown lookup value"

### ISSUE THREAD

**Issue by jeriox** [abrensch/brouter#463]
I am trying to modify the rail profile present in brouter-web to allow routing via funiculars. OSM specifies railway=funicular for that purpose (see https://wiki.openstreetmap.org/wiki/Tag:railway%3Dfunicular). When modifing the profile to include this tag, brouter returns `unknown lookup value: funicular`. As this is a valid OSM tag, I expect brouter to allow for routes to be planned with this tag.

**Comment by jeriox** [comment]
thanks for the quick response and the fix!

**Comment by MaartenDeen** [comment]
Has something gone wrong with this commit? It is still not in the master branch.

I would really like for it to be added.

**Comment by flecken66** [comment]
I've noticed that besides funicular, monorail and miniature are also missing and cannot be routed.

It would be nice if these two types could also be added:
Monorail: `railway=monorail` https://wiki.openstreetmap.org/wiki/Tag:railway%3Dmonorail
Miniature: `railway=miniature` https://wiki.openstreetmap.org/wiki/Tag:railway%3Dminiature

**Comment by afischerdev** [comment]
@flecken66 
After a quick search in Germany, I only found Wuppertal and possibly Frankfurt Airport. There may be more that are connected to public transportation and have transitions to other public paths.
But most of them were in fun parks with loops.
Sorry, but I don't see much need for them.

**Comment by flecken66** [comment]
@afischerdev 
I understand your point regarding the low number of relevant monorails in Germany. (In addition, there is the H-Bahn in Dortmund, the SkyTrain in Düsseldorf and Schwebebahn in Dresden.)

While this might not justify changing the default railway profile, I believe users who write their own custom profiles should at least have the option to allow routing over these systems. (So adding them to the lookups.dat file would allow this.)

Furthermore, looking beyond Europe, I would argue that monorail systems do have a relevance in daily life as a serious public transport option. https://en.wikipedia.org/wiki/List_of_monorail_systems

This ensures BRouter remains useful for users mapping railway travel in Germany and in the world.

**Comment by afischerdev** [comment]
@flecken66 
> I believe users who write their own custom profiles should at least have the option to allow routing over these systems

Yes, that sounds good. But it's not just an entry in the lookups.dat file. It's also about the size of data files. And this includes many useless paths.
The list also contains amusement parks.

**Comment by flecken66** [comment]
I didn't know that adding tags to the lookups.dat file would affect more than just that file and impact the total size. To see how much data this involves, I ran a worldwide Overpass query for both `railway=monorail` and `railway=miniature`. The resulting .osm export file is only ∼20 MB.
Given the minimal size of this global data, I would assume that adding both tags would have no noticeable negative impact on brouter's performance or file size. I think the slight overhead from potentially unnecessary paths is a fair trade-off for giving advanced users the flexibility to map and route across all modes of transport.

As a compromise, if the 20 MB is still considered too much, focusing only on monorails would result in a file size of just 6 MB. This option would also eliminate the vast majority of miniature railways that only run in small loops as an attraction. (I know it would still contain monorails in amusement parks though. There is no "solution" for that.)

**Comment by afischerdev** [comment]
@flecken66 
Ok, the size argument is gone. You are right and I could live with the 'monorails only' solution.
Please give me some time, I like to test what happens to isolated monorails like [this](https://www.openstreetmap.org/way/108235602#map=19/49.036503/9.052186) or [this](https://www.openstreetmap.org/way/389370167#map=19/48.821352/9.228726) inside BRouter data. I could start in the end of next week.

**Comment by flecken66** [comment]
@afischerdev 
I appreciate you considering adding monorails. Please take all the time to test the cases with those isolated monorails. There's absolutely no need to rush.

​Thank you so much for your great work on this project!

**Comment by afischerdev** [comment]
I did some test.
The monorails could be connected.

<img width="273" height="267" alt="Image" src="https://github.com/user-attachments/assets/9b3a6a29-9971-4feb-ace0-7b6f867a743d" />

And we have around 300 in Germany.
